Source/Core/Common/Config/ConfigInfo.h

```diff
@@ -18,36 +18,43 @@ template <typename T>
 using UnderlyingType = typename std::enable_if_t<std::is_enum<T>{}, std::underlying_type<T>>::type;
 }  // namespace detail
 
-struct Location
-{
-  System system;
-  std::string section;
-  std::string key;
-
-  bool operator==(const Location& other) const;
-  bool operator!=(const Location& other) const;
-  bool operator<(const Location& other) const;
-};
-
 template <typename T>
-struct Info
+class Info
 {
-  Info(const Location& location_, const T& default_value_)
-      : location{location_}, default_value{default_value_}
+  friend class ActiveInfos<T>;
+
+public:
+  Info(const Location& location, const T& default_value)
+      : m_location{location}, m_default_value{default_value}
   {
+    ActiveInfos<T>::Update(this);
+    ActiveInfos<T>::Add(this);
   }
 
   // Make it easy to convert Info<Enum> into Info<UnderlyingType<Enum>>
   // so that enum settings can still easily work with code that doesn't care about the enum values.
   template <typename Enum,
             std::enable_if_t<std::is_same<T, detail::UnderlyingType<Enum>>::value>* = nullptr>
   Info(const Info<Enum>& other)
-      : location{other.location}, default_value{static_cast<detail::UnderlyingType<Enum>>(
-                                      other.default_value)}
+      : m_location{other.GetLocation()}, m_default_value{static_cast<detail::UnderlyingType<Enum>>(
+                                             other.GetDefaultValue())}
   {
+    ActiveInfos<T>::Update(this);
+    ActiveInfos<T>::Add(this);
   }
 
-  Location location;
-  T default_value;
+  Info(const Info<T>& other) = default;
+  Info(Info<T>&& other) = delete;
```

Comment:
The assignment operators should be visibly defaulted or deleted to make the assignment semantics explicit as well.